### DISCUSSION TITLE
Relative paths for markdown files used inside a React app.

### DISCUSSION THREAD

**Discussion by prakhathi-m** [remarkjs/remark#1119] [Q&A] [answered]
Hi,

I'm trying to use `react-markdown` in my react app which is built with webpack. I used `raw-loader` in the webpack config to load md files. 

webpack.config.js:
```
{
        test: /\.md$/, 
        use: ['raw-loader']
},
```

Markdown.js

```
import ReactMarkdown from 'react-markdown';
import File from './docs/file.md';

ReactDom.render(
  <ReactMarkdown children={File}  />,
  document.body
)
```

File.md
`[click here for relative path file](./test.md)`

I added multiple files inside docs folder. File.md has couple of information and links to these other files. 

I would see File.md rendered in the app. However, If I click on the link to test.md, I get `Cannot GET /test.md` .
I know I have imported only File.md and passed it to the component. But, Is there an example where this is achieved?
or any plugins that I can use to navigate through relative path links?

Thank you.

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @prakhathi-m! 👋 
Sorry you ran into a spot of trouble.

It sounds like your end goal is to have a directory of static markdown files and be able to navigate between them as pages?
If so I'd highly recommend using one of the following:

* https://nextjs.org/
* https://astro.build/
* https://remix.run/
* https://qwik.builder.io/qwikcity/overview/
* https://www.gatsbyjs.com/

Which are design to support this very use case.
And include routers which can navigate between markdown files, and resolve relative links.

---

To answer your direct question, if you're hand rolling a build system.
You'll also need to hand roll a virtual file system/router.
`react-markdown` will render your markdown document, but it is not a router.
You would need to create a router to handle loading the correct markdown file, that is not something `react-markdown` can or will do.

If you need to tweak relative links to work with your custom routing system, `remark` plugins can help with that, you can draw inspiration from https://github.com/zslabs/remark-relative-links and https://github.com/GaiAma/Coding4GaiAma/tree/HEAD/packages/remark-truncate-links

Again, I'd recommend taking a look at one of the pre-built solutions listed above, routing has a lot of complexity and edge cases which need to be considered, all of the above have already put a lot of hard work into solving those problems for you.

**Comment by prakhathi-m** [comment]
Thank you @ChristianMurphy for the quick response. I figured routing is the way to go here. Resolved this issue. Thanks for your help.
